node/src/perf_config.rs

```diff
@@ -0,0 +1,96 @@
+use std::collections::HashMap;
+
+use graph::{anyhow, prelude::regex::Regex};
+use graph_core::{SubgraphPerfConfig, SubgraphPerfRules};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct SubgraphConfigSection {
+    rules: HashMap<String, SubgraphConfigRule>,
```

Comment:
I don't think that will help much since the main issue is that usually people would prolly be expecting a first match and a catch-all rule. Internally, the consistent ordering is not super important (I think).